includes/modules/sideboxes/shopping_cart.php

```diff
@@ -8,19 +8,25 @@
  * @version $Id: DrByte 2020 Jul 10 Modified in v1.5.8-alpha $
  */
 
+  $gv_balance = 0.00;
+  if (zen_is_logged_in() && !zen_in_guest_checkout()) {
+    $customer = new Customer;
```

Comment:
i am a fan of line 13, as well as now removing the call to the function `zen_user_has_gv_account` in this script.

if merged, said function will now only have 1 call in the base code, which is in the script `tpl_checkout_payment_default.php`.

given that we are now moving towards a more class based ideology (at least from what i am seeing), i do not understand how this `$customer` object is not available during the checkout payment process.  i would think that if a customer is logged in, we should have a `$customer` object which is an instance of the `Customer` class.

i think eliminating calls to these `zen_` functions would make zc much more flexible; and needless to say i am a fan.

i suppose what i am getting at, is that the conditional for line 12 should be done much sooner during the load process.  if the customer is logged in, why do we not create the Customer object at an earlier time and make it much more available?

Reply:
Done